app/src/forms/common/models/models.js

```diff
@@ -335,16 +341,44 @@ class UserFormRoles extends Model {
           query.where('formName', 'ilike', `%${value}%`);
         }
       },
-      filterPublic(query, value) {
-        if (value !== undefined) {
-          query.where('public', value);
-        }
-      },
       filterActive(query, value) {
         if (value !== undefined) {
           query.where('active', value);
         }
       },
+      filterByAccess(query, idps, roles, permissions) {
+        if (idps || roles || permissions) {
+          const toArray = (values) => {
+            if (values) {
+              return Array.isArray(values) ? values.filter(p => p && p.trim().length > 0) : [values].filter(p => p && p.trim().length > 0);
+            }
+            return [];
+          };
+          const inArrayClause = (column, values) => {
+            return values.map(p =>`'${p}' = ANY("${column}")`).join(' or ');
+          };
+          const _idps = toArray(idps);
+          const _roles = toArray(roles);
+          const _permissions = toArray(permissions);
+          let clauses = [];
+
+          if (_idps.length) {
+            clauses.push(`('{}'::varchar[] = "roles" and (${inArrayClause('idps', _idps)}))`);
+          }
+
+          if (_roles.length) {
+            clauses.push(`(array_length("roles", 1) > 0 and (${inArrayClause('roles', _roles)}))`);
+          }
+
+          if (_permissions.length) {
+            clauses.push(`(array_length("permissions", 1) > 0 and (${inArrayClause('permissions', _permissions)}))`);
+          }
+
+          if (clauses.length) {
+            query.whereRaw(`(${clauses.join(' or ')})`);
+          }
+        }
+      },
```

Comment:
Do we need to consider some kind of return value when neither idps, roles nor permissions are specified? I think the return value of this function would be undefined as it is atm.

Reply:
There is no return in these methods.  They build out the database query. If there are no criteria, then we don't alter the query (add where clause etc).